### ISSUE TITLE
Significant Difference On ASR Between My Experiment And Original Paper

### ISSUE THREAD

**Issue by clee-jaist** [hkunzhe/label_consistent_attacks_pytorch#4]
Hi Author,

The original paper reported  that the training effect of directly adding trigger is not good without PGD attack on the targeted class of data, but after my test, it is totally differnent. Since the original paper only trained 100 epochs, but I found that the first 100 epochs fluctuated greatly (I trained 200 epochs), and there were  epochs with ASR lower than 60% (consistent with the paper: the cat class is in poison rate=25 %, the ASR did not exceed 60%), but the ASR after 150 epoch was basically stable at around 91%. This confuse me, is it true that the LC attack is only effective when training for 100 epochs? When gradually increasing the training epoch, it is equally effective to directly add trigger training to the targeted class image without using LC attack?. Below are the last three epochs where the trigger was added directly to the cat image for training.

Thank you so much for your time.

---------------------------------------------------------------------------------------------------------------------------------------------------
===Epoch: 198/200===
Poison training...
Training summary:
|---------|---------|----------|
|    loss |     acc |     time |
|---------|---------|----------|
| 0.00123 | 0.99994 | 12.03157 |
Test model on clean data...
|---------|---------|---------|
|    loss |     acc |    time |
|---------|---------|---------|
| 0.21658 | 0.94818 | 0.86664 |
Test model on poison data...
|---------|---------|---------|
|    loss |     acc |    time |
|---------|---------|---------|
| 0.22935 | 0.91521 | 0.82752 |
Saved model for epoch 198 (overwritten)
Adjust learning rate to 0.0010000000000000002
===Epoch: 199/200===
Poison training...
Training summary:
|---------|---------|----------|
|    loss |     acc |     time |
|---------|---------|----------|
| 0.00116 | 1.00000 | 12.27233 |
Test model on clean data...
|---------|---------|---------|
|    loss |     acc |    time |
|---------|---------|---------|
| 0.21673 | 0.94818 | 0.88241 |
Test model on poison data...
|---------|---------|---------|
|    loss |     acc |    time |
|---------|---------|---------|
| 0.22477 | 0.91730 | 0.80264 |
Saved model for epoch 199 (overwritten)
Adjust learning rate to 0.0010000000000000002
===Epoch: 200/200===
Poison training...
Training summary:
|---------|---------|----------|
|    loss |     acc |     time |
|---------|---------|----------|
| 0.00111 | 1.00000 | 11.91589 |
Test model on clean data...
|---------|---------|---------|
|    loss |     acc |    time |
|---------|---------|---------|
| 0.22030 | 0.94818 | 0.86948 |
Test model on poison data...
|---------|---------|---------|
|    loss |     acc |    time |
|---------|---------|---------|
| 0.22866 | 0.91655 | 0.81496 |
Saved model for epoch 200 (overwritten)
Adjust learning rate to 0.0010000000000000002
Best poison test accuracy achieved at epoch 149: 96.22%

Process finished with exit code 0

**Comment by hkunzhe** [comment]
1. Firstly, I am not the author of the original LC paper. The motivation I created this repo is that the original paper code was based on TensorFlow. However, I wanted to provide a minimal PyTorch implementation. Therefore, while ensuring sufficiently high ASR, **some settings are not consistent with the original paper**. These differences are mentioned in the Readme, and some training settings (such as the learning rate and its scheduler) need to be compared with the details of the code and paper.
2. Considering your question, from my experience, the main reason is probably due to the optimizer used in LC training being SGD, and its learning rate decays at 0.1 after 100 and 150 iterations respectively. Therefore, ASR only starts to converge after 100 epochs.
